File: src/main/java/no/nav/domain/pensjon/regler/beregning/BeregningRelasjon.java
package no.nav.domain.pensjon.regler.beregning;

import no.nav.domain.pensjon.regler.beregning2011.Beregning2011;
import no.nav.domain.pensjon.regler.beregning2011.Uforetrygdberegning;

import java.io.Serializable;
import java.lang.reflect.Constructor;
import java.lang.reflect.InvocationTargetException;

/**
 * @author Aasmund Nordstoga (Acenture) - PKFEIL-2605
 */
public class BeregningRelasjon implements Serializable {

    /**
     * 1967 beregningen som det relateres til
     */
    private Beregning beregning;

    /**
     * Beregning 2011 som det relateres til
     */
    private Beregning2011 beregning2011;

    /**
     * Angir om beregningen er brukt (helt eller delvis) i beregningen den tilhører.
     */
    private boolean bruk;

}

File: src/main/java/no/nav/pensjon/regler/domain/beregning/BeregningRelasjon.kt
package no.nav.pensjon.regler.domain.beregning

import no.nav.pensjon.regler.domain.beregning2011.Beregning2011
import java.io.Serializable

class BeregningRelasjon : Serializable {
    /**
     * 1967 beregningen som det relateres til
     */
    var beregning: Beregning? = null

    /**
     * Beregning 2011 som det relateres til
     */
    var beregning2011: Beregning2011? = null

    /**
     * Angir om beregningen er brukt (helt eller delvis) i beregningen den tilhører.
     */
    var isBruk = false

    /**
     * Referanse tilbake til beregning hvor beregningsrelasjon inngår i delberegningslisten.
     */
    var pREG_parentBeregning: Beregning? = null

    /**
     * Referanse tilbake til beregning2011 hvor beregningsrelasjon inngår i delberegningslisten.
     */
    var pREG_parentBeregning2011: Beregning2011? = null
}